src/Microsoft.DotNet.XHarness.Apple/AppRunnerBase.cs

```diff
@@ -76,9 +91,18 @@ protected async Task<string> FindDevice(TestTargetOs target)
 
             var envVars = environmentVariables.ToDictionary(
                 p => p.Key,
-                p => p.Value is bool ? p.Value.ToString().ToLowerInvariant() : p.Value.ToString()); // turn True to true
+                p => p.Value is bool ? p.Value.ToString().ToLowerInvariant() : p.Value.ToString()); // turns "True" to "true"
 
-            return _processManager.ExecuteCommandAsync(binaryPath, arguments, _mainLog, timeout, envVars, cancellationToken);
+            systemLog.StartCapture();
+
+            try
+            {
+                return await _processManager.ExecuteCommandAsync(binaryPath, arguments, _mainLog, timeout, envVars, cancellationToken);
+            }
+            finally
+            {
+                systemLog.StopCapture();
```

Comment:
Does it stop logging in any case?

Reply:
What do you mean "any case"? Even when there's an exception? (then the answer would be yes)

Reply:
No, the opposite case - without exception

Reply:
Yes, finally should do it in all cases